python/GafferTest/ContextTest.py

```diff
@@ -81,10 +81,9 @@ def f( context, name ) :
 		hash3 = c.hash()
 		self.assertEqual( changes, [ ( "a", 2, hash1 ), ( "a", 3, hash2 ), ( "b", 1, hash3 ) ] )
 
-		# when an assignment makes no actual change, the signal should not
-		# be triggered again.
+		# Even an assignment that doesn't change anything does still trigger the signal
```

Comment:
In the case of a pointer based set, I don't think we can get it right, since we probably want to allow:
```int i = 1;
s.set( "foo", &i );
i = 2;
s.set( "foo", &i );```
And the second set should trigger a changedSignal.  I figured it was probably better to be consistent, and assume we never skip changedSignal.

Reply:
Makes sense for pointer-based set, but those are only on `EditableScope`, where we never expect `m_changedSignal` to even be created. How hard would it be to support for the non-pointer case, without adding overhead to the pointer case? Happy to deal with this as a followup if that's the best way of keeping this moving.